### ISSUE TITLE
Rules: Cannot update color value

### ISSUE THREAD

**Issue by jaller94** [mozilla-iot/gateway#1266]
### Steps to Reproduce
0. Enable virtual devices and add a Virtual Color Light
1. Go into the Rules
2. Add a rule
3. Switch from `While` to `If` (**This part got added**)
4. Add the light as your output
5. Select its property color
6. Change the color

### Expected
The rule’s description gets updated and the change gets saved.

### Actual
While the browser’s color picker opens and the input accepts new colors, the gateway will ignore those changes. The description of the rule will keep the first selected color. When leaving and re-entering the rule the old color will be selected again.

![screenshot from 2018-08-12 02-18-16](https://user-images.githubusercontent.com/10872136/44000521-0ea6b2a8-9dd6-11e8-9eb8-defc94b98111.png)
Note that the color is now red, but the description still says `#ffffff`.

**Comment by benfrancis** [comment]
Thanks for this bug report, I'm not sure I'm quite understanding the steps to reproduce because I can't reproduce the bug. The changes do seem to take place.

What browser and OS are you using?

@hobinjk are you able to reproduce this?

**Comment by hobinjk** [comment]
Yeah, I followed the exact steps and could not reproduce the incorrect behavior.

**Comment by jaller94** [comment]
I am using the Mozilla Things 0.5.0 stnadard release for the Raspberry Pi.
It had been upgraded from the 0.4.1 image.

I was able to reproduce this with
* Firefox stable 61 on Ubuntu
* Firefox Beta 62 on Android
* Chrome 68 on Android

**Comment by jaller94** [comment]
Sorry. The steps to reproduce this must be incomplete. I still have rules from 2 days ago which are affected by this, but new rules I created today are not affected.

I am using the Mozilla Things 0.5.0 standard release for the Raspberry Pi.
It had been upgraded from the 0.4.1 image.

I was able to reproduce this with exitings rules on
* Firefox stable 61 on Ubuntu
* Firefox Beta 62 on Android
* Chrome 68 on Android
